### ISSUE TITLE
Numba Reduction operations (+=, *=) not supported on GPU

### ISSUE THREAD

**Issue by adarshyoga** [IntelPython/numba-dpex#71]
Add-assign, multiply-assign operations within Numba par-fors which are reduction operations need to be supported on GPU.

Here is an example kernel in pygbm where "+=" reduction operation is necessary.

``` python
@njit(parallel=True)
def _update_raw_predictions(leaves_data, raw_predictions):
  for leaf_idx in prange(len(leaves_data)):
     leaf_value, sample_indices = leaves_data[leaf_idx]
     for sample_idx in sample_indices:
         raw_predictions[sample_idx] += leaf_value

```

**Comment by diptorupd** [comment]
TODOs to get prange reductions for `+` and `*` working in main:

- [ ] Update #999 on top of current main
- [ ] Fix SLM constant size validation pass
- [ ] Turn on  block renaming and remove dels options in gufunc kernel generation
- [ ] Review #999 and do any further refactoring as needed (code quality)
- [ ] Add unit tests

**Comment by diptorupd** [comment]
Fixed in #999
